Search, Canonical Pages URL, Metadata fixes

fullreboot opened the pull request:
-Fixes Search
-Cleans up controller code
-Fixes Pages canonical url (/posts/page-title --> /page-title)

Notes for future: getBlogPosts() and getSingleBlogPost() Method should be moved to domain if possible.

Commit:
Refactored Home controller to get posts from common method

-Fixes Search
-Cleans up controller code

Notes for future: getBlogPosts() Method should be moved to domain if possible. DTO object needed?

Commit:
fixes pages canonical url

Commit:
Small fixes to fix routing of /admin

Commit:
Links Fix

ghost commented:
There's still an open bug in this pull request:

- Route conflict with `HttpGet["{slug}"]` and serving the static file index.html for the route `/admin` or `/admin/`

Commit:
MVC Catch-All Route Fix for /admin Pager

Added controller with explicit route /admin to catch request to admin page and serve proper index.html. Previously this route was being incorrectly handled by the /{slug} controller action

ghost commented:
MVC Catch-All error has been fixed. While implementing the new pages canonical urls from `/pages/{slug}` to `/{slug}` the admin route `/admin` was being improperly intercepted.

Fixed in this commit which was automatically added to the PR

https://github.com/fullreboot/Blogifier/commit/5a178696014e75e30a61184ce88385c10ce73cbd

> Added controller with explicit route /admin to catch request to admin page and serve proper index.html. Previously this route was being incorrectly handled by the /{slug} controller action

Commit:
Pr Cleanup. Cleaning uneeded changes, mostly URLS

Commit:
Remove unnecessary Controller Action parameter
